client/app/bundles/course/assessment/containers/AssessmentForm.jsx

```diff
@@ -0,0 +1,223 @@
+        <Field
+          name="password_protected"
+          component={Toggle}
+          label={<FormattedMessage {...translations.passwordProtection} />}
+          labelPosition="right"
+          disabled={submitting}
+        />
+        {
+          this.props.password_protected &&
+          <Field
+            name="password"
+            component={TextField}
+            hintText={<FormattedMessage {...translations.password} />}
+            fullWidth
+            autoComplete={false}
+            disabled={submitting}
+          />
+        }
+        <div style={styles.hint}>
+          <FormattedMessage {...translations.passwordProtectionHint} />
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    const { handleSubmit, onSubmit, submitting } = this.props;
+
+    return (
+      <Form onSubmit={handleSubmit(onSubmit)}>
+        <Field
+          name="title"
+          component={TextField}
+          floatingLabelText={<FormattedMessage {...translations.title} />}
+          style={styles.title}
+          disabled={submitting}
+        />
+        <br />
+        <Field
+          name="description"
+          component={RichTextField}
+          label={<FormattedMessage {...translations.description} />}
+          style={styles.description}
+          disabled={submitting}
+        />
+        <div style={styles.flexGroup}>
+          <Field
+            name="start_at"
+            component={DateTimePicker}
+            floatingLabelText={<FormattedMessage {...translations.startAt} />}
+            style={styles.flexChild}
+            disabled={submitting}
+          />
+          <Field
+            name="end_at"
+            component={DateTimePicker}
+            floatingLabelText={<FormattedMessage {...translations.endAt} />}
+            style={styles.flexChild}
+            disabled={submitting}
+          />
+          <Field
+            name="bonus_end_at"
+            component={DateTimePicker}
+            floatingLabelText={<FormattedMessage {...translations.bonusEndAt} />}
+            style={styles.flexChild}
+            disabled={submitting}
+          />
+        </div>
+        <div style={styles.flexGroup}>
+          <Field
+            name="base_exp"
+            component={TextField}
+            floatingLabelText={<FormattedMessage {...translations.baseExp} />}
+            type="number"
+            style={styles.flexChild}
+            disabled={submitting}
+          />
+          <Field
+            name="time_bonus_exp"
+            component={TextField}
+            floatingLabelText={<FormattedMessage {...translations.timeBonusExp} />}
+            type="number"
+            style={styles.flexChild}
+            disabled={submitting}
+          />
+        </div>
+        <Field
+          name="autograded"
+          component={Toggle}
+          label={<FormattedMessage {...translations.autograded} />}
+          labelPosition="right"
+          style={{ marginTop: 24 }}
```

Comment:
I think this is okay. shall we standardize that if there is only one style and the style is not shared, we keep it inline instead of putting it in `styles`?